File: src/main/java/net/minestom/server/listener/SettingsListener.java
package net.minestom.server.listener;

import net.minestom.server.entity.Player;
import net.minestom.server.event.EventDispatcher;
import net.minestom.server.event.player.PlayerSettingsChangeEvent;
import net.minestom.server.network.packet.client.play.ClientSettingsPacket;

public final class SettingsListener {
    public static void listener(ClientSettingsPacket packet, Player player) {
        Player.PlayerSettings settings = player.getSettings();
        final byte viewDistance = (byte) Math.abs(packet.viewDistance());
        settings.refresh(packet.locale(), viewDistance, packet.chatMessageType(), packet.chatColors(), packet.displayedSkinParts(), packet.mainHand(), packet.enableTextFiltering(), packet.allowsListing());
        EventDispatcher.call(new PlayerSettingsChangeEvent(player));
    }
}

File: src/main/java/net/minestom/server/listener/SettingsListener.kt
package net.minestom.server.listener

import net.minestom.server.entity.Player
import net.minestom.server.network.packet.client.play.ClientSettingsPacket
import net.minestom.server.entity.Player.PlayerSettings
import net.minestom.server.event.EventDispatcher
import net.minestom.server.event.player.PlayerSettingsChangeEvent

object SettingsListener {
    fun listener(packet: ClientSettingsPacket, player: Player) {
        val settings = player.settings
        val viewDistance = Math.abs(packet.viewDistance().toInt()).toByte()
        settings.refresh(
            packet.locale(),
            viewDistance,
            packet.chatMessageType(),
            packet.chatColors(),
            packet.displayedSkinParts(),
            packet.mainHand(),
            packet.enableTextFiltering(),
            packet.allowsListing()
        )
        EventDispatcher.call(PlayerSettingsChangeEvent(player))
    }
}